packages/webpack/src/config/base.js

```diff
@@ -57,10 +58,13 @@ export default class WebpackBaseConfig {
   normalizeTranspile () {
     // include SFCs in node_modules
     const items = [/\.vue\.js/i]
-    for (const pattern of this.buildContext.buildOptions.transpile) {
+    for (let pattern of this.buildContext.buildOptions.transpile) {
+      if (typeof pattern === 'function') {
+        pattern = pattern(this.nuxtEnv)
+      }
       if (pattern instanceof RegExp) {
         items.push(pattern)
-      } else {
+      } else if (pattern) {
```

Comment:
Should this be string ?

Reply:
Yes, Will update to check `typeof pattern === 'string'`